next/app/lib/types/User.ts

```diff
@@ -0,0 +1,4 @@
+export interface User {
```

Comment:
I don't think you need this type; the `User` type used in the Navbar component comes from the @auth/core module (node_modules/@auth/core/src/types.ts), and that type already has the `name` field. It was also augmented (https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation) with the custom User type defined in next/types/next-auth.ts, which has the `roles` field.

Also, I think if you base this PR off the `develop` branch, you'll see that the typescript errors in the Navbar component associated with `user` have been fixed already!